ocaml/lambda/lambda.ml

```diff
@@ -1392,16 +1395,25 @@ let mod_field ?(read_semantics=Reads_agree) pos =
 let mod_setfield pos =
   Psetfield (pos, Pointer, Root_initialization)
 
+(* [primitive_may_allocate] treats projecting an unboxed float from a float
+   record as non-allocating, which is a lie for the bytecode backend (where
+   unboxed floats are boxed).  But it's only used for stack allocation, which
+   doesn't happen in bytecode.  If that changes, or if we want to use this for
+   another purpose in bytecode, it will need to be revised.
+*)
```

Comment:
I think this comment should be in the `.mli`, so you see it when using the function (also you could make it a doc comment there so it will appear in editor popups).